openwisp_firmware_upgrader/admin.py

```diff
@@ -200,22 +204,65 @@ def __get_rate(self, value):
     aborted_rate.short_description = _('abortion rate')
 
 
+class DeviceFirmwareForm(forms.ModelForm):
+    class Meta:
+        model = load_model('DeviceFirmware')
+        fields = '__all__'
+
+    def _get_compatible_boards(self, instance):
+        # instance.device.model in instance.image.boards
+        # boards = FIRMWARE_IMAGE_MAP[instance.image.type]['boards']
+        device_model = instance.device.model
+        qs = load_model('FirmwareImage').objects.filter(
+            build__category__organization=instance.device.organization
+        )  # Or filter by type=...
+        list_of_ids = [
```

Comment:
I would also take in consideration the category.
Something like the following should work very well:

if the DeviceFirmware instance already exists, select images of the same category used by the current firmware, this will make upgrading very easy.
when the DeviceFirmware instance does not exist, show all categories.